handling-forms/submitting-forms.md

```diff
@@ -0,0 +1,428 @@
+
+<br/>
+
+If we want to change the name of the expected keys, we should refactor `App`'s definition of `addStudentData`.
+
+### !end-callout
+
+Now, Sofia needs to actually use her new function. She passes it into an instance of `NewStudentForm`, through a new prop named `addStudentCallback`.
+
+<!-- prettier-ignore-start -->
+```js
+    return (
+        <main>
+            <h1>Attendance</h1>
+            <StudentList
+                students={studentData}
+                onUpdateStudent={updateStudentData}
+            ></StudentList>
+            <NewStudentForm
+                addStudentCallback={addStudentData}
+            ></NewStudentForm>
+        </main>
+    );
+```
+<!-- prettier-ignore-end -->
+
+Sofia updates the PropTypes of `NewStudentForm` to now anticipate this prop:
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+
+const NewStudentForm = (props) => {
+// ...
+
+NewStudentForm.propTypes = {
+    addStudentCallback: PropTypes.func
+};
+```
+<!-- prettier-ignore-end -->
+
+## Handling Form Submissions 
+
+Sofia's next step is to get `NewStudentForm` to add student data when the form submits.
+
+She starts by creating a new function, `onFormSubmit`:
+
+<!-- prettier-ignore-start -->
+```js
+    const onFormSubmit = (event) => {
+        event.preventDefault();
+
+        props.addStudentCallback({
+            nameData: formFields.name,
+            emailData: formFields.email
+        });
+
+        setFormFields({
+            name: '',
+            email: '',
+        });
+    };
+```
+<!-- prettier-ignore-end -->
+
+Sofia's implementation of `onFormSubmit` looks like this:
+
+1. She uses the passed in `event` object and calls `event.preventDefault()`. This prevents the unwanted default behavior of HTML forms.
+1. She invokes the function `props.addStudentCallback()`. This prop was passed in by `App`, with the value `addStudentData`.
+1. She knows that `props.addStudentCallback` receives an object, `newStudent`. She knows that this object should have the keys `nameData` and `emailData`. She passes in an object literal, where the keys are `nameData` and `emailData`, and the values read from the `formFields` state.
+1. She resets the form by updating the members of `formFields` to empty strings.
+
+### !callout-danger
+
+## Prevent the Form's Default Behavior
+
+HTML forms have default behavior: when a form receives a "submit" event, it will make a GET request. This creates an effect where our web app reloads every time we submit a form! Be sure to include `event.preventDefault();` in any form submission event handler.
+
+### !end-callout
+
+## Verify
+
+Sofia's app now handles the new student form submission!
+
+![](../assets/handling-forms_submitting-forms_newstudentform-callback-diagram.png)
+
+Her last step is to verify her work.
```

Comment:
with the above suggestions, now it's been a bit since Sofia's name, so bring Sofia back here
```suggestion
Sofia's last step is to verify her work.
```